src/transport_ofi.c

```diff
@@ -297,7 +297,15 @@ static inline int allocate_endpoints(struct fabric_info *info)
      * support shmem_wait that blocks on an update arriving from the network */
 #ifndef ENABLE_HARD_POLLING
     info->p_info->caps |= FI_REMOTE_WRITE | FI_REMOTE_READ;
+#else
+    info->p_info->caps &= ~FI_REMOTE_WRITE;
+    info->p_info->caps &= ~FI_REMOTE_READ;
 #endif
+
+    /* Remove unused capabilities */
+    info->p_info->caps &= ~FI_MSG;
+    info->p_info->caps &= ~FI_TAGGED;
```

Comment:
OTOH, some providers might just provide you with more functionality than you asked for. No harm in turning them off again.